Packages/CandidateUI/Sources/CandidateUI/VerticalCandidateController.swift

```diff
@@ -286,17 +286,22 @@ public class VerticalCandidateController: CandidateController {
                 return
             }
 
-            var lastVisibleRow = newValue
-
-            if selectedRow != -1 && itemCount > 0 && itemCount > labelCount {
-                if newIndex > selectedRow && (Int(newIndex) - selectedRow) > 1 {
-                    lastVisibleRow = min(newIndex + UInt(labelCount) - 1, itemCount - 1)
+            if itemCount > labelCount {
+                var rowToScroll = Int(newValue)
+
+                if selectedRow != -1 && itemCount > 0 {
+                    let firstVisibleRow = tableView.row(at: scrollView.documentVisibleRect.origin)
+                    if newIndex > selectedRow && (Int(newIndex) - selectedRow) > 1 {
```

Comment:
Could you also use the opportunity to comment on why we also need to check `(Int(newIndex) - selectedRow) > 1`, because we don't want to do this adjustment if it's only single row movement (i.e. when the Down key is pressed).